File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()){
            return false;
        }
        unordered_map<char,int> m;
        for(int i=0;i<s.size();i++){
            m[s[i]]++;
        }
        for(int i=0;i<t.size();i++){
            if(m.find(s[i])==m.end()||m[t[i]]==0){
            return false;
            }
            m[t[i]]--;

        }

        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        count_s = {}
        count_t = {}
        for i in range(len(s)):
            value_s = s[i]
            value_t = t[i]

            if value_s in count_s:
                count_s[value_s] += 1
            else:
                count_s[value_s] = 1

            if value_t in count_t:
                count_t[value_t] += 1
            else:
                count_t[value_t] = 1
        
        return count_s == count_t

            


# TC = O(N)
# SC = O(1)